<a href="https://colab.research.google.com/github/husnullkhatimah/Proyek-Data-Wrangling/blob/main/Tugas%201%20Data%20Wrangling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Nama   : Husnul Khatimah
## Nim    : 241061017
## Prodi  : Sains Data

In [1]:
import pandas as pd
df = pd.read_csv('/content/heart.csv')
df

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1020,59,1,1,140,221,0,1,164,1,0.0,2,0,2,1
1021,60,1,0,125,258,0,0,141,1,2.8,1,1,3,0
1022,47,1,0,110,275,0,0,118,1,1.0,1,1,2,0
1023,50,0,0,110,254,0,0,159,0,0.0,2,0,2,1


In [3]:
df.head()


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0




*   Dataset ini merupakan data klasifikasi kesehatan jantung yang terdiri dari 14 kolom dan cuplikan 5 baris pertama. Secara teknis, seluruh data disajikan dalam bentuk numerik (angka), dengan mayoritas bertipe integer dan satu kolom (oldpeak) bertipe float atau desimal.
*   Isi utamanya terbagi menjadi tiga bagian: identitas pasien (usia dan jenis kelamin), indikator klinis (seperti tekanan darah, kolesterol, dan detak jantung), serta variabel target (kolom target) yang menunjukkan status kesehatan pasien. Sebagian besar angka dalam kolom seperti sex, cp, dan fbs sebenarnya adalah data kategorikal yang sudah dikodekan menjadi angka untuk memudahkan proses hitungan mesin atau algoritma.






In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1025 entries, 0 to 1024
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1025 non-null   int64  
 1   sex       1025 non-null   int64  
 2   cp        1025 non-null   int64  
 3   trestbps  1025 non-null   int64  
 4   chol      1025 non-null   int64  
 5   fbs       1025 non-null   int64  
 6   restecg   1025 non-null   int64  
 7   thalach   1025 non-null   int64  
 8   exang     1025 non-null   int64  
 9   oldpeak   1025 non-null   float64
 10  slope     1025 non-null   int64  
 11  ca        1025 non-null   int64  
 12  thal      1025 non-null   int64  
 13  target    1025 non-null   int64  
dtypes: float64(1), int64(13)
memory usage: 112.2 KB




*   Dataset ini memiliki total 1.025 baris data (entri) yang terindeks dari 0 hingga 1024. Terdapat 14 kolom informasi yang tersedia secara keseluruhan. Dari sisi kualitas data, seluruh kolom memiliki status 1025 non-null, yang berarti tidak terdapat data yang kosong atau hilang (missing values) pada setiap barisnya.
*   Terkait tipe data, dataset ini didominasi oleh bilangan bulat (int64) sebanyak 13 kolom, sementara satu kolom lainnya, yaitu oldpeak, menggunakan tipe data bilangan desimal (float64). Secara teknis, seluruh dataset ini tersimpan dengan efisien dalam memori dengan penggunaan kapasitas sekitar 112.2 KB.



In [6]:
df.shape

(1025, 14)


*   Berdasarkan informasi dari perintah df.info() dan df.shape yang Anda lampirkan, dataset ini memiliki struktur yang sangat rapi dengan total 1.025 baris (entri) dan 14 kolom data. Seluruh kolom dalam dataset ini bersifat "lengkap" karena memiliki 1025 non-null count, yang berarti tidak terdapat data yang kosong atau hilang (missing values) pada setiap variabelnya.
*   Dari sisi teknis, dataset ini hampir sepenuhnya terdiri dari bilangan bulat dengan 13 kolom bertipe int64, sementara hanya terdapat satu kolom, yaitu oldpeak, yang menggunakan tipe data bilangan desimal atau float64. Secara keseluruhan, data ini disimpan secara efisien dalam memori dengan penggunaan kapasitas sekitar 112.2 KB.



In [7]:
import json
df = [
    {
        "id": 1,
        "nama": "Andi Saputra",
        "usia": 45,
        "jenis_kelamin": "Pria",
        "tekanan_darah": 130,
        "kolesterol": 240,
        "merokok": True,
        "gejala": ["nyeri dada", "sesak nafas"],
        "diagnosis": "Positif",
        "tanggal_periksa": "2025-02-15"
    },
    {
        "id": 2,
        "nama": "Siti Aminah",
        "usia": 52,
        "jenis_kelamin": "Wanita",
        "tekanan_darah": 145,
        "kolesterol": 280,
        "merokok": False,
        "gejala": ["lelah", "pusing"],
        "diagnosis": "Negatif",
        "tanggal_periksa": "2025-02-16"
    },
    {
        "id": 3,
        "nama": "Budi Santoso",
        "usia": 38,
        "jenis_kelamin": "Pria",
        "tekanan_darah": 120,
        "kolesterol": 200,
        "merokok": True,
        "gejala": ["jantung berdebar"],
        "diagnosis": "Positif",
        "tanggal_periksa": "2025-02-17"
    },
    {
        "id": 4,
        "nama": "Dewi Lestari",
        "usia": 61,
        "jenis_kelamin": "Wanita",
        "tekanan_darah": 148,
        "kolesterol": 203,
        "merokok": False,
        "gejala": ["nyeri dada", "lelah", "sesak"],
        "diagnosis": "Positif",
        "tanggal_periksa": "2025-02-18"
    },
    {
        "id": 5,
        "nama": "Rudi Hermawan",
        "usia": 29,
        "jenis_kelamin": "Pria",
        "tekanan_darah": 118,
        "kolesterol": 182,
        "merokok": False,
        "gejala": [],
        "diagnosis": "Negatif",
        "tanggal_periksa": "2025-02-19"
    }
]
with open('data_pasien.json', 'w') as f:
    json.dump(df, f, indent=2)
df

[{'id': 1,
  'nama': 'Andi Saputra',
  'usia': 45,
  'jenis_kelamin': 'Pria',
  'tekanan_darah': 130,
  'kolesterol': 240,
  'merokok': True,
  'gejala': ['nyeri dada', 'sesak nafas'],
  'diagnosis': 'Positif',
  'tanggal_periksa': '2025-02-15'},
 {'id': 2,
  'nama': 'Siti Aminah',
  'usia': 52,
  'jenis_kelamin': 'Wanita',
  'tekanan_darah': 145,
  'kolesterol': 280,
  'merokok': False,
  'gejala': ['lelah', 'pusing'],
  'diagnosis': 'Negatif',
  'tanggal_periksa': '2025-02-16'},
 {'id': 3,
  'nama': 'Budi Santoso',
  'usia': 38,
  'jenis_kelamin': 'Pria',
  'tekanan_darah': 120,
  'kolesterol': 200,
  'merokok': True,
  'gejala': ['jantung berdebar'],
  'diagnosis': 'Positif',
  'tanggal_periksa': '2025-02-17'},
 {'id': 4,
  'nama': 'Dewi Lestari',
  'usia': 61,
  'jenis_kelamin': 'Wanita',
  'tekanan_darah': 148,
  'kolesterol': 203,
  'merokok': False,
  'gejala': ['nyeri dada', 'lelah', 'sesak'],
  'diagnosis': 'Positif',
  'tanggal_periksa': '2025-02-18'},
 {'id': 5,
  'nama': 'R

In [12]:
df.to_csv("data_pasien.csv", index=False)
df

,id,nama,usia,jenis_kelamin,tekanan_darah,kolesterol,merokok,gejala,diagnosis,tanggal_periksa
0,1,Andi Saputra,45,Pria,130,240,True,"[nyeri dada, sesak nafas]",Positif,2025-02-15
1,2,Siti Aminah,52,Wanita,145,280,False,"[lelah, pusing]",Negatif,2025-02-16
2,3,Budi Santoso,38,Pria,120,200,True,[jantung berdebar],Positif,2025-02-17
3,4,Dewi Lestari,61,Wanita,148,203,False,"[nyeri dada, lelah, sesak]",Positif,2025-02-18
4,5,Rudi Hermawan,29,Pria,118,182,False,[],Negatif,2025-02-19
